octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/DescriptionComposite.java

```diff
@@ -1,69 +1,160 @@
+        stackLayoutComposite.showControl(browserDescHtml);
+    }
+
+    public void setEntityModel(EntityModelWrapper entityModelWrapper) {
+        browserDescHtml.setText(getBrowserText(entityModelWrapper.getReadOnlyEntityModel()));
+    }
+
+    public String getBrowserText(EntityModel entityModel) {
+        String descriptionFromServerRemodeled = Util.getUiDataFromModel(entityModel.getValue((EntityFieldsConstants.FIELD_DESCRIPTION)));
+        try {
+            descriptionFromServerRemodeled = downloadPictures(
+                    Util.getUiDataFromModel(entityModel.getValue((EntityFieldsConstants.FIELD_DESCRIPTION))));
+        } catch (IOException e) {
+        }
+
+        String descriptionText = "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">"
+                + descriptionFromServerRemodeled
+                + "</font></body></html>";
+        if (descriptionText.equals("<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">" + "</font></body></html>")) {
+
+            return "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                    + getRgbString(foregroundColor) + ">" + "No description" + "</font></body></html>";
+        } else {
+            return descriptionText;
+        }
+
+    }
+
+    private static String getRgbString(Color color) {
+        return "rgb(" + color.getRed() + "," + color.getGreen() + "," + color.getBlue() + ")";
+    }
+
+    private String downloadPictures(String descriptionField) throws IOException {
+        String tmpPath = System.getProperty("java.io.tmpdir");
+        String baseUrl = Activator.getConnectionSettings().getBaseUrl();
+        File tmpDir = new File(tmpPath);
+        octanePhotosDir = new File(tmpDir, "Octane_pictures");
+
+        if (!new File(tmpDir, "Octane_pictures").exists()) {
+            octanePhotosDir.mkdir();
+        }
+        String octanePhotosPath = octanePhotosDir.getAbsolutePath();
+
+        Document descriptionParser = Jsoup.parse(descriptionField);
+        Elements link = descriptionParser.getElementsByTag("img");
+
+        for (Element el : link) {
+            String pictureLink = el.attr("src");
+            if (pictureLink.contains("/api/shared_spaces")) {
+                el.attr("src", baseUrl + pictureLink);
+                pictureLink = el.attr("src");
+            }
+
+            if (!pictureLink.contains(Activator.getConnectionSettings().getBaseUrl())) {
+                continue;
+            }
+            int index = pictureLink.lastIndexOf("/");
+            String pictureName = pictureLink.substring(index + 1, pictureLink.length());
+            String picturePath = octanePhotosPath + "\\" + pictureName;
+            if (!new File(octanePhotosDir, pictureName).exists()) {
```

Comment:
@vladbenchea I'm assuming you have this check so that you don't download the same image multiple times. But have you considered the following cases: 
1) can different pictures have the same name? Both in the same workspace and in different workspaces
2) you downloaded picture with name "a.jpg" and then the user deleted "a.jpg" from octane and uploaded a new picture also called "a.jpg"

Reply:
1. I assumed the fact that the server where getting the images from won't have 2 imagines with the same name. Never actually tried it
2. That's a case I should discuss with Tibi, didn't think about it. Thanks!